google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITDatabaseTest.java

```diff
@@ -43,4 +54,68 @@ public void badDdl() {
 
     env.getTestHelper().createTestDatabase("CREATE TABLE T ( Illegal Way To Define A Table )");
   }
+
+  @Test
+  public void databaseDeletedTest() throws Exception {
+    // Create a test db, do a query, then delete it and verify that it returns
+    // DatabaseNotFoundExceptions.
+    Database db = env.getTestHelper().createTestDatabase();
+    DatabaseClient client = env.getTestHelper().getClient().getDatabaseClient(db.getId());
+    try (ResultSet rs = client.singleUse().executeQuery(Statement.of("SELECT 1"))) {
+      assertThat(rs.next()).isTrue();
+      assertThat(rs.getLong(0)).isEqualTo(1L);
+      assertThat(rs.next()).isFalse();
+    }
+
+    // Delete the database.
+    db.drop();
+    // We need to wait a little before Spanner actually starts sending DatabaseNotFound errors.
+    ExponentialBackOff backoff =
+        new ExponentialBackOff.Builder()
+            .setInitialIntervalMillis(1000)
+            .setMaxElapsedTimeMillis(35000)
+            .setMaxIntervalMillis(5000)
+            .build();
+    DatabaseNotFoundException notFoundException = null;
+    long millis = 0L;
+    while ((millis = backoff.nextBackOffMillis()) != ExponentialBackOff.STOP) {
+      Thread.sleep(millis);
+      // Queries to this database should eventually return DatabaseNotFoundExceptions.
+      try (ResultSet rs = client.singleUse().executeQuery(Statement.of("SELECT 1"))) {
+        rs.next();
+      } catch (DatabaseNotFoundException e) {
+        // This is what we expect.
+        notFoundException = e;
```

Comment:
I guess we should break the `while` here once we get the error.

Reply:
Yep, good catch.